src/main/kotlin/org/rust/lang/core/psi/RsFile.kt

```diff
@@ -165,9 +169,14 @@ class RsFile(
         // This occurs if the file is not included to the project's module structure, i.e. it's
         // most parent module is not mentioned in the `Cargo.toml` as a crate root of some target
         // (usually lib.rs or main.rs). It's anyway useful to know cargoProject&workspace in this case
-        val cargoProject = project.cargoProjects.findProjectForFile(virtualFile) ?: return EMPTY_CACHED_DATA
-        val workspace = cargoProject.workspace ?: return CachedData(cargoProject)
-        return CachedData(cargoProject, workspace)
+        val stdlibCrates = project.crateGraph.topSortedCrates
+            .filter { it.origin == PackageOrigin.STDLIB }
```

Comment:
Since stdlib crates are always first in the top sorted list, we can write it this way:
```suggestion
            .takeWhile { it.origin == PackageOrigin.STDLIB }
```

Reply:
There are also `STDLIB_DEPENDENCY` so can't use `takeWhile`

Reply:
Then it could be `.takeWhile { it.origin == PackageOrigin.STDLIB || it.origin == PackageOrigin.STDLIB_DEPENDENCY }` :D